services/issue/pull.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package issue
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	issues_model "code.gitea.io/gitea/models/issues"
+	org_model "code.gitea.io/gitea/models/organization"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/gitrepo"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+func getMergeBase(repo *git.Repository, baseBranch, headBranch string) (string, error) {
+	// Add a temporary remote
+	tmpRemote := strconv.FormatInt(time.Now().UnixNano(), 10)
```

Comment:
It's better to have a proper prefix for the temp remote name

Reply:
d6b195eb45ec1e299c29c5da78af869d7c33f605